### DISCUSSION TITLE
Content Security Policy (CSP)

### DISCUSSION THREAD

**Discussion by hoang1998lqd** [jhipster/generator-jhipster#27594] [Q&A]
I am having trouble configuring Content Security Policy (CSP) at the gateway when not using 'unsafe-eval' 'unsafe-inline' in the configuration.

**Comment by mraible** [comment]
Please don't cross-post as an issue and a discussion, we see both. I've added a comment to [your issue](https://github.com/jhipster/generator-jhipster/issues/27595).

**Comment by TatyOko28** [comment]
This is a common issue when hardening CSP in a JHipster gateway.

By default, removing:

```text
'unsafe-inline'
'unsafe-eval'

**Comment by leverglowh** [comment]
All issues on CSP are marked as stale and closed, but the generator still ships with those directives and inline js in `index.html` and `swagger-ui/index.html`.
I believe we need to move the code to separate `.js` files to safely remove `unsafe-inline`.
Not sure why `unsafe-eval` is needed, some dev tools would be blocked without it, so it's probably useful in development.

https://github.com/jhipster/generator-jhipster/blob/0830cc9049386c6fdf92a15e0cc76b97a2b8df1e/generators/client/templates/src/main/webapp/index.html.ejs#L121-L132
